Updated and verified the MacOS installation guide (#7)
203-Systems/Matrix-Wiki: docs/Developer/BuildingMatrixOS/Mystrix/MacOS.md

1. Visit [https://brew.sh/](https://brew.sh/) and follow the instructions to install the latest version of Homebrew
2. After installation, you may need to follow any on-screen instructions to add Homebrew to your system's PATH. Once installed, verify it by running:

```bash
brew --version
```

## Install Dependencies
The Matrix OS codebase is hosted in a Git repository, it also uses the Espressif IoT Development Framework for building and uploading to the Mystrix device. We'll install the dependencies for both by running:

```bash
brew install git cmake ninja dfu-util python3
pip3 install psutil
```

## Clone Matrix OS repo
1. Clone the Matrix OS repository:
   ```bash
   git clone https://github.com/203-Systems/MatrixOS.git
   ```

2. Navigate to the cloned repository:
   ```bash
   cd MatrixOS
   ```

3. Initialize the submodules in the Matrix OS repository:
   ```bash
   git submodule update --init
   ```

## Install ESP IDF v5.3.1
Run the following to set up the ESP IDF
```bash
mkdir -p ~/esp
cd ~/esp

> git clone -b v5.4.1 --recursive https://github.com/espressif/esp-idf.git

Reviewer: ```suggestion
git clone -b v5.3.1 --recursive https://github.com/espressif/esp-idf.git
```

# Building on macOS

## Installing Homebrew
Homebrew is a popular package manager for macOS and Linux, allowing you to install and manage software efficiently from the command line. We will use it to install the dependencies needed to build Matrix OS.

1. Visit [https://brew.sh/](https://brew.sh/) and follow the instructions to install the latest version of Homebrew
2. After installation, you may need to follow any on-screen instructions to add Homebrew to your system's PATH. Once installed, verify it by running:

```bash
brew --version
```

## Install Dependencies
The Matrix OS codebase is hosted in a Git repository, it also uses the Espressif IoT Development Framework for building and uploading to the Mystrix device. We'll install the dependencies for both by running:

```bash
brew install git cmake ninja dfu-util python3
pip3 install psutil
```

## Clone Matrix OS repo
1. Clone the Matrix OS repository:
   ```bash
   git clone https://github.com/203-Systems/MatrixOS.git
   ```

2. Navigate to the cloned repository:
   ```bash
   cd MatrixOS
   ```

3. Initialize the submodules in the Matrix OS repository:
   ```bash
   git submodule update --init
   ```

> ## Install ESP IDF v5.4.1

Reviewer: Matrix OS is still on ESP IDF 5.3.x. 5.4 might works but I haven't reviewed changes.

Run the following to set up the ESP IDF
```bash
mkdir -p ~/esp
cd ~/esp
git clone -b v5.4.1 --recursive https://github.com/espressif/esp-idf.git
cd ~/esp/esp-idf
./install.sh esp32
python3 $HOME/esp/esp-idf/tools/idf_tools.py install riscv32-esp-elf
```

## Build Matrix OS
1. Load ESP-IDF by sourcing it in your terminal session. You can run this command to set up ESP-IDF for the session:
   ```bash
   source ~/esp/esp-idf/export.sh
   ```

>    In the long run, you will want to automate this. You could add this line to your shell's configuration file (e.g., `.zshrc` or `.bash_profile`) or if you are using VS Code, you can modify the MatrixOS.code-workspace file and adapt it to run it on new terminal. I use `.zshrc` and add `alias get_idf='. $HOME/esp/esp-idf/export.sh'`. When I want to do Matrix OS development I run `get_idf`

Reviewer: Better drop first person wording.

Reviewer: ```suggestion
   In the long run, you will want to automate this. You could add this line to your shell's configuration file (e.g., `.zshrc` or `.bash_profile`) or if you are using VS Code, you can modify the MatrixOS.code-workspace file and adapt it to run it on new terminal. You can also use `.zshrc` and add `alias get_idf='. $HOME/esp/esp-idf/export.sh'`, then you can run `get_idf` to get esp idf in your environment.
```

2. Go to the root folder of Matrix OS if your terminal isn’t already there.
3. Run this command to build Matrix OS:
   ```bash
   make DEVICE=Mystrix build
   ```

>    If you run into any undefined references, make sure to check the `fullclean` command in the "Build Commands" section below

Reviewer: This only happens after files are modified around. Better put this in make your own apps instead on initial build guide

Reviewer: ```suggestion
```
